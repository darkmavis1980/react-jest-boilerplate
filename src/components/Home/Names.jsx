import React from 'react';
import fetchData from '../../api/fetchData.js';

const resource = fetchData('/sample.json');

const Names = () => {
  const namesList = resource.read();

  return (
    <div>
      <h2>List of names</h2>
      <p>This component will use a custom handler for fetching data.</p>
      <ul>
        {namesList.map(item => (
          <li key={item.id}>
            {item.name}
          </li>))}
      </ul>
    </div>
  );
};

export default Names;
