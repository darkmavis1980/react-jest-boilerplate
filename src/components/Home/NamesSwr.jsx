import React from 'react';
import useSWR from 'swr';
import axios from 'axios';

const fetcher = url => axios.get(url).then(({data}) => data);

const NamesSwr = () => {
  const { data: namesList } = useSWR('/sample.json', fetcher, { suspense: true});

  return (
    <div>
      <h2>List of names with SWR</h2>
      <p>This component will use the <a href="https://swr.vercel.app/" target="_blank" rel="noreferrer">SWR hook</a> for fetching data.</p>
      <ul>
        {namesList.map(item => (
          <li key={item.id}>
            {item.name}
          </li>))}
      </ul>
    </div>
  );
};

export default NamesSwr;
