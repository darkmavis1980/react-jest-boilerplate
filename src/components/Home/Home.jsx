import React, { Suspense } from 'react';
import Names from './Names';
import NamesSwr from './NamesSwr';
import ErrorBoundary from '../ErrorBoundary';
import Loading from '../Loading';

const Home = () => (
  <div>
    <h1>Hello World</h1>
    <ErrorBoundary fallback={<p>Could not load the data from the API.</p>}>
      <Suspense fallback={<Loading />}>
        <Names />
      </Suspense>
    </ErrorBoundary>
    <ErrorBoundary fallback={<p>Could not load the data from the API.</p>}>
      <hr />
      <Suspense fallback={<Loading />}>
        <NamesSwr />
      </Suspense>
    </ErrorBoundary>
  </div>
);

export default Home;
