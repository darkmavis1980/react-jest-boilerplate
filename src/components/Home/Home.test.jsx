import React from 'react';
import { create } from 'react-test-renderer';
import Home from './Home';

describe('Home (Component)', () => {
  it('should match the snapshot', async () => {
    const tree = await create(<Home />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render the hello world title', async () => {
    const wrapper = await create(<Home />);
    const instance = wrapper.root;
    expect(instance.findByType('h1').children).toContain('Hello World');
  });
});
