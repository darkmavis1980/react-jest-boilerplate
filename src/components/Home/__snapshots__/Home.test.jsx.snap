// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home (Component) should match the snapshot 1`] = `
<div>
  <h1>
    Hello World
  </h1>
  <p>
    Loading...
  </p>
  <hr />
  <p>
    Loading...
  </p>
</div>
`;
